### PR TITLE
Library refactor

### PR DESCRIPTION
**Pull request by LightOfHeaven1994**
Cards that are going to be closed:
- [IQE-2342](https://issues.redhat.com/browse/IQE-2342) Stop depend on iqe-jenkins repository
- [IQE-2343](https://issues.redhat.com/browse/IQE-2343) Remove parameters and reload logic from pipeline library
- [IQE-2345](https://issues.redhat.com/browse/IQE-2345) Make UI and xdist flags false by default
- [IQE-2346](https://issues.redhat.com/browse/IQE-2346) Add option to override default user in execIQETests
- [IQE-2396](https://issues.redhat.com/browse/IQE-2396) Add possibility to provide success pipeline run message

Change list:
- Remove config logic and stop depending on iqe-jenkins repo
- Remove obsolete code
- Support forceDefaultUser to handle user selection
- Add ability to define success message in execIQETestsWithNotifier
- Refactor execIQETestsWithNotifier helper methods
- Correctly return previous build to define if pipeline status changed
- Add piece of doc about using execIQETests

### CODE REVIEW

**Comment by bsquizz** [comment]
I'll double check to make sure that the old frontend build jobs won't be
impacted by anything here.

**Comment by LightOfHeaven1994** [comment]
@bsquizz sure, thank you! I can ping you after it's ready for the review. I still need to do some adjustments, then going to test it with multiple apps on CSB.

File: vars/openShiftUtils.groovy
Comment (bsquizz) [inline vars/openShiftUtils.groovy:69]: This doesn't need to block the MR, but perhaps moving to v6 is a good time to rename all occurences of `jenkinsSlave` to `jenkinsAgent`?

**Review by bsquizz — APPROVED** [review]
LGTM! Only way to know if the re-work of the test failure/resolve notification is good is by testing :) I'd say you're free to refactor this library to however you see fit now w/ the migration to CSB. What you have here looks good to me. I confimed that the legacy FE build jobs are pinned to `v3` so merging these changes shouldn't impact those jobs. This is the list of repos I can see in our org that have NOT pinned their import of the library: https://github.com/search?q=org%3ARedHatInsights+%40Library%28%22github.com%2FRedHatInsights%2Finsights-pipeline-lib+NOT+%40Library%28%22github.com%2FRedHatInsights%2Finsights-pipeline-lib%40&type=code -- I think a lot of this is likely "stale" and not high-impact -- so I would just send an email about the upcoming change a couple days before you merge this into `master` to let these teams know.

**Review by san7ket — APPROVED** [review]
LGTM!! tagging it, should help with gradually moving over to v6

**Comment by LightOfHeaven1994** [comment]
Sent an email about the change, going to merge & ask someone to release v6 **next week**

**Pull request merged**
